src/vs/workbench/contrib/debug/browser/disassemblyView.ts

```diff
@@ -522,23 +592,68 @@ class InstructionRenderer extends Disposable implements ITableRenderer<IDisassem
 	}
 
 	renderTemplate(container: HTMLElement): IInstructionColumnTemplateData {
+		const sourcecode = append(container, $('.sourcecode'));
 		const instruction = append(container, $('.instruction'));
+		this.applyFontInfo(sourcecode);
 		this.applyFontInfo(instruction);
-
 		const currentElement: { element?: IDisassembledInstructionEntry } = { element: undefined };
 
 		const disposables = [
-			this._disassemblyView.onDidChangeStackFrame(() => this.rerenderBackground(instruction, currentElement.element))
+			this._disassemblyView.onDidChangeStackFrame(() => this.rerenderBackground(instruction, sourcecode, currentElement.element)),
+			addStandardDisposableListener(sourcecode, 'click', () => this.openSourceCode(currentElement.element?.instruction!)),
 		];
 
-		return { currentElement, instruction, disposables };
+		return { currentElement, instruction, sourcecode, disposables };
 	}
 
 	renderElement(element: IDisassembledInstructionEntry, index: number, templateData: IInstructionColumnTemplateData, height: number | undefined): void {
-		templateData.currentElement.element = element;
+		this.renderElementInner(element, index, templateData, height);
+	}
 
+	private async renderElementInner(element: IDisassembledInstructionEntry, index: number, templateData: IInstructionColumnTemplateData, height: number | undefined): Promise<void> {
+		templateData.currentElement.element = element;
 		const instruction = element.instruction;
-		const sb = createStringBuilder(10000);
+		templateData.sourcecode.innerText = '';
+		const sb = createStringBuilder(1000);
+
+		if (this._disassemblyView.isSourceCodeRender && instruction.location?.path && instruction.line) {
+			let sourceURI = this.getUriFromSource(instruction);
+
+			if (sourceURI) {
+				let textModel: ITextModel | undefined = undefined;
+				const sourceSB = createStringBuilder(10000);
+
+				try {
+					const ref = await this._disassemblyView.textModelService.createModelReference(sourceURI);
+					textModel = ref.object.textEditorModel;
+					templateData.disposables.push(ref);
```

Comment:
Loos like ref is not used anymore after getting the content of some source. So instead of inserting this to disposables, consider disposing it at the end of this function. If we should keep it whenever this line is in view, you can dispose it in disposeElement().

Reply:
Not disposing will allow the file to persist when rending another source block of the same file.  Granted, this could lead to 100s of files opened, but that should be rare.

Reply:
@xisui-MSFT , Can you take care of this change when you refactor to listview?

Reply:
@roblourens This is intended. Otherwise the listeners would be created/disposed too frequently #129527. 

Refs to files shouldn't be kept across sessions though IMO. But as Felix says, the cases where there are too many files opened should be rare.